vllm/compilation/collective_fusion.py

```diff
@@ -131,8 +131,10 @@ def __init__(self, config: VllmConfig):
         AllGatherGEMMPattern(self.model_dtype,
                              self.device).register(self.patterns)
 
-    def is_applicable_for_shape(self, shape: Optional[int]) -> bool:
-        # only do replace for specific shapes
+    def is_applicable(self, splitting_ops: list[str],
+                      shape: Optional[int]) -> bool:
+        if splitting_ops is None or splitting_ops == []:
```

Comment:
I think let's save this from config during init so we're not adding extra params to this function